### ISSUE TITLE
Implement context manager in ChunkingZipFile.readfile() zip file entries

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8641]
|[<img alt="scop's avatar" src="https://avatars.githubusercontent.com/u/109152?s=50" width="50" height="50">](https://github.com/scop)| @scop reported|
|-|-|
|Trac ID|trac#8641|
|Type|enhancement|
|Created|2016-07-20 18:40:57Z|

<details><summary>Searchable metadata</summary>

```
trac-id__8641 8641
type__enhancement enhancement
reporter__scop scop
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469040057355013 1469040057355013
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="scop's avatar" src="https://avatars.githubusercontent.com/u/109152?s=50" width="50" height="50">](https://github.com/scop)<a name="note_1"></a>|@scop commented|
|-|-|

https://github.com/twisted/twisted/pull/371

**Comment by twisted-trac** [comment]
|<img alt="Ville Skyttä's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|Ville Skyttä commented|
|-|-|

In changeset 7f019becd343797d327ea73650bb36f68fe09f35
```
#!CommitTicketReference repository="" revision="7f019becd343797d327ea73650bb36f68fe09f35"
Add topfile for #8641
```

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset ade1f34319a53f118e5dbb2d41e5844052bba387
```
#!CommitTicketReference repository="" revision="ade1f34319a53f118e5dbb2d41e5844052bba387"
Merge pull request #371 from scop/8641-scop-zfe-ctxmgr

Author: scop
Reviewer: rodrigc
Fixes: #8641

Implement context manager in ChunkingZipFile.readfile() zip file entries
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
